src/payment/strategies/braintree/braintree-payment-processor.ts

```diff
@@ -133,6 +133,10 @@ export default class BraintreePaymentProcessor {
             throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
         }
 
+        if (typeof nonce === 'undefined' || !nonce) {
```

Comment:
Why not just `if (!nonce) {` ? 🤔

Reply:
Also, you can use line 132 `if (!this._threeDSecureOptions || !nonce) {`

Reply:
> Why not just `if (!nonce) {` ? 🤔

because in typescript a string can be undefined too

Reply:
> Also, you can use line 132 `if (!this._threeDSecureOptions || !nonce) {`

I could merge that code